### DISCUSSION TITLE
How do I use Class property in XAML?

### DISCUSSION THREAD

**Discussion by SF-Simon** [dotnet/maui#14135] [Q&A]
BasePage.xaml
```
<?xml version="1.0" encoding="UTF-8" ?>
<ContentPage
    x:Class="App.Views.Bases.BasePage"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:local="clr-namespace:App.Views.Bases">
    <ContentPage.Resources>
        <ResourceDictionary>
            <Thickness
                x:Key="AdaptiveBoundaryTest"
                Bottom="{Binding AdaptiveBoundary.Bottom, Mode=OneWay}"
                Left="{Binding AdaptiveBoundary.Left, Mode=OneWay}"
                Right="{Binding AdaptiveBoundary.Right, Mode=OneWay}"
                Top="{Binding AdaptiveBoundary.Top, Mode=OneWay}" />
        </ResourceDictionary>
    </ContentPage.Resources>
    <ContentPage.Content />
</ContentPage>
```

BasePage.xaml.cs
```
namespace iWriting3.Views.Bases
{
	public partial class BasePage : ContentPage
	{
		public Thickness AdaptiveBoundaryProperty { get; private set; } = new Thickness(12, 0);

		public BasePage()
		{
			InitializeComponent();
			this.BindingContext = this;
		}
	}
}
```

I want to create a base class that has some properties that all ContentPages can inherit and use. These attributes are dynamic and not necessarily immutable.

But when I wrote this, it reported a mistake.

`XFC0009 No property, BindableProperty, or event found for "Bottom", or mismatching type between value and property.`

What should I write about this process? 

thx.

**Comment by codeaphex** [comment]
I think the problem is the "wrong" usage of resource dictionaries.

1. Problem: The resource dictionary has no knowledge where it should bind too.
It is possible to bind to a relative source, see  
https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/data-binding/relative-bindings?view=net-maui-7.0  
but you need probably switch to a controltemplate inside the ResourceDictionary  
https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/data-binding/relative-bindings?view=net-maui-7.0#bind-to-a-templated-parent
This is just for reference I think the 2. solution is more appropiate for your usage, 

2. Problem: The resource dictionary would be duplicated multiple times
If you would implement it like now, this resource dictionary would be duplicated for each page inheriting the base.
You could try setting up the resource somewhere higher up the visual layout, like on app level, to use this as a base with hardcoded values. If you need other values on a page add the resource again directly on the page and set the values like needed:
See https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/resource-dictionaries?view=net-maui-7.0#override-resources
